crates/ekore/src/anomalous_dimensions/unpolarized/spacelike/as4/ggg.rs

```diff
@@ -0,0 +1,201 @@
+            - 437084.0 * (-(1. / (-1. + n).powu(2)) + 1. / n.powu(2))
+            - 361570.0 * 1. / ((-1. + n) * n)
+            + 1696070.0 * (1. / (n + n.powu(2)))
+            - 1457385.0 * (1. / (2. + 3. * n + n.powu(2)))
+            + 3195104.0 * (-(1. / n.powu(2)) + 1. / (1. + n).powu(2))
+            + 2009021.0 * 2. / n.powu(3)
+            + 544380.0 * (-(6. / n.powu(4)))
+            + 9938.0 * lm11m1(n, S1)
+            + 24376.0 * lm12m1(n, S1, S2)
+            - 22143.0 * lm13m2(n, S1, S2, S3);
+        P3ggApp2 = P3gg01
+            - 706649.0 * (-(1. / (-1. + n).powu(2)) + 1. / n.powu(2))
+            - 2274637.0 * 1. / ((-1. + n) * n)
+            + 836544.0 * (1. / n - n / (2. + 3. * n + n.powu(2)))
+            - 199929.0 * (1. / (6. + 5. * n + n.powu(2)))
+            - 2683760.0 * (-(1. / n.powu(2)) + 1. / (1. + n).powu(2))
+            + 168802.0 * 2. / n.powu(3)
+            - 250799.0 * (-(6. / n.powu(4)))
+            + 36967.0 * lm11m1(n, S1)
+            + 24530.0 * lm12m1(n, S1, S2)
+            - 71470.0 * lm12m2(n, S1, S2);
+    } else if nf == 5 {
+        P3ggApp1 = P3gg01
+            - 439426.0 * (-(1. / (-1. + n).powu(2)) + 1. / n.powu(2))
+            - 293679.0 * 1. / ((-1. + n) * n)
+            + 1916281.0 * (1. / (n + n.powu(2)))
+            - 1615883.0 * (1. / (2. + 3. * n + n.powu(2)))
+            + 3648786.0 * (-(1. / n.powu(2)) + 1. / (1. + n).powu(2))
+            + 2166231.0 * 2. / n.powu(3)
+            + 594588.0 * (-(6. / n.powu(4)))
+            + 50406.0 * lm11m1(n, S1)
+            + 24692.0 * lm12m1(n, S1, S2)
+            + 174067.0 * lm11m2(n, S1);
+        P3ggApp2 = P3gg01
+            - 705978.0 * (-(1. / (-1. + n).powu(2)) + 1. / n.powu(2))
+            - 2192234.0 * 1. / ((-1. + n) * n)
+            + 1730508.0 * (1. / (2. + 3. * n + n.powu(2)))
+            + 353143.0
+                * ((12. + 9. * n + n.powu(2))
+                    / (6. * n + 11. * n.powu(2) + 6. * n.powu(3) + n.powu(4)))
+            - 2602682.0 * (-(1. / n.powu(2)) + 1. / (1. + n).powu(2))
+            + 178960.0 * 2. / n.powu(3)
+            - 218133.0 * (-(6. / n.powu(4)))
+            + 2285.0 * lm11m1(n, S1)
+            + 19295.0 * lm12m1(n, S1, S2)
+            - 13719.0 * lm12m2(n, S1, S2);
+    } else {
+        panic!("nf=6 is not available at N3LO");
+    }
+
+    //We return (for now) one of the two error-band representatives
+    //or the present best estimate, their average
+    let P3GGA = Complex::from(match variation {
+        1 => P3ggApp1,
+        2 => P3ggApp2,
+        _ => 0.5 * (P3ggApp1 + P3ggApp2),
+    });
+    -P3GGA
+}
+
+#[cfg(test)]
+mod tests {
+
+    use super::*;
+    use crate::harmonics::cache::Cache;
+    use crate::{assert_approx_eq_cmplx, cmplx};
+    use num::complex::Complex;
+
+    #[test]
+    fn test_reference_moments() {
+        fn gg3_moment(N: usize, nf: f64) -> f64 {
+            let nf2 = nf * nf;
+            let nf3 = nf2 * nf;
+            let mom_list = [
```

Comment:
please add reference: in this case I guess it is Eq. xy of [blub]

Reply:
done in b0869d11